docs/bls_key_password_management.md

```diff
@@ -0,0 +1,111 @@
+# BLS Key Management in Babylon
+
+This document describes how BLS keys are managed in Babylon, including password
+management options and command-line flags for secure key generation and usage.
+
+## Overview
+
+BLS keys in Babylon are used for validator signatures in the checkpointing
+mechanism. These keys are stored encrypted using the ERC-2335 keystore format
+and require a password for decryption when the node starts.
+
+## Password Management Options
+
+There are three ways to provide the password for the BLS key:
+
+1. **Environment Variable (Recommended)**: Set the `BABYLON_BLS_PASSWORD` 
+   environment variable
+2. **Password File**: Store the password in a file (default location: 
+   `$HOME/.babylond/config/bls_password.txt` or custom location)
+3. **Interactive Prompt**: Enter the password when prompted during node startup
+
+## Key Generation
+
+When generating a new BLS key (during `init` or using `create-bls-key`), you'll
+be prompted to choose how to manage your password:
+
+```
+Select the storage strategy for your BLS password.
+1. Environment variable (recommended)
```

Comment:
hmm this is one thing I am not sure about. I.e when creating new key and entering passphrase for the first time it is usually up to user to store it somewhere. I think this is unexpected that creation / init command tries to store it either in env var or file. 

So ideally creation command would only show the prompt to enter new passphrase and after that it would not store it anywhere else. Of course, there should be some warning that user must save this passphrase somewhere or remember it.

Then when starting node there are at lest 2 options:
1. (recommended) read from env variable
2. (not recomended but possible when using explicit flag) from file

Do you see some problem with this flow ?

Reply:
Makes sense then I guess what's better is to just print out some instructions after the password prompt on what is recommended and takes precedent (env over file path) and rely on them setting it or providing a path to a file ?

Reply:
I would say so.

In my experience, I have never had neither wallet nor node save my passphrase anywhere by default, so this would be surprising behaviour which could lead to some problems.

It is usually up to operator, that he has his passphrase ready and can use it later when necessary.